File: JavaServer/DdzServer/src/net/silverfoxserver/extmodel/RoomModelByToac.java
/*
 * SilverFoxServer: massive multiplayer game server for Flash, ...
 * VERSION:3.0
 * PUBLISH DATE:2015-9-2 
 * GITHUB:github.com/wdmir/521266750_qq_com
 * UPDATES AND DOCUMENTATION AT: http://www.silverfoxserver.net
 * COPYRIGHT 2009-2015 SilverFoxServer.NET. All rights reserved.
 * MAIL:521266750@qq.com
 */
package net.silverfoxserver.extmodel;

import net.silverfoxserver.core.model.IUserModel;
import net.silverfoxserver.core.util.RandomUtil;
import net.silverfoxserver.DdzLogic_Toac;

/**
 *
 * @author ACER-FX
 */
public class RoomModelByToac {
    
    /** 
	 
    */
    private IUserModel _u;

    public final IUserModel getUser()
    {
            return _u;
    }

    /** 

    */
    private String[] _pickResult = {"0", "0", "0"};

    /** 
     winOrLost,fen,machine_pick
    */

    public final String[] getpickResult()
    {

            return _pickResult;

    }


    /** 


     @param value
    */
    public RoomModelByToac(IUserModel value)
    {
            _u = value.clone();

            _pickResult = new String[]{"0", "0", "0"};
    }

    /** 
     默认2倍，闪光为3倍
     我压中的概率是三分之一

     @param selectInd_
     @param fen
     @return 
    */
    public final String[] pick_a_card(int selectInd_, long jiaoFen)
    {
            //winOrLost,fen,machine_pick,your_pick,light
            //string[] pickResult = { "0", "0", "0" };
            _pickResult = new String[] {"0", "0", "0", "0","0"};

            //
            java.util.Random r = new java.util.Random(RandomUtil.GetRandSeed());

            int machine_pick = r.nextInt(3);
            machine_pick++;

            int light = r.nextInt(2);

            //
            _pickResult[2] = (new Integer(machine_pick)).toString();
            _pickResult[3] = (new Integer(selectInd_)).toString();
            _pickResult[4] = (new Integer(light)).toString();

            //
            if (selectInd_ == machine_pick)
            {
                    _pickResult[0] = "1";

                    //获得的分值 = 押分;
                    if (0 == light)
                    {
                            _pickResult[1] = String.valueOf(jiaoFen * 1);
                    }
                    else
                    {
                            _pickResult[1] = String.valueOf(jiaoFen * 2);
                    }

            }
            else
            {
                    _pickResult[0] = "0";

                    //扣掉的分值 = 押分 * 1
                    _pickResult[1] = String.valueOf(jiaoFen * 1);
            }

            return _pickResult;
    }


    /** 


     @param n
     @param v
     @return 
    */
    public final String[] setVars(String n, String v)
    {

            String[] setStatus = {"false", "", ""};

            //
            String[] sp = v.split("[,]", -1);


            //selectInd
            int selectInd = Integer.parseInt(sp[1]);
            long jiaoFen = Long.parseLong(sp[2]);

            //
            if (1 != selectInd && 2 != selectInd && 3 != selectInd)
            {
                    setStatus[0] = "false";
                    setStatus[1] = "<status>1</status>";

                    return setStatus;

            }

            //
            if (DdzLogic_Toac.getg1() != jiaoFen && 
                DdzLogic_Toac.getg2() != jiaoFen && 
                DdzLogic_Toac.getg3() != jiaoFen)
            {
                    setStatus[0] = "false";
                    setStatus[1] = "<status>2</status>";

                    return setStatus;
            }

            //---------------------------------------------------
            pick_a_card(selectInd, jiaoFen);

            //
            setStatus[0] = "true";
            setStatus[1] = _pickResult[0] + "," + _pickResult[1] + "," + _pickResult[2];

            return setStatus;
    }

    /** 


     @return 
    */
    public final String toXMLString()
    {

            StringBuilder sb = new StringBuilder();

            //
            sb.append("<module>");
            sb.append("<m n='");
            sb.append(DdzLogic_Toac.name);
            sb.append("'>");

            sb.append("<pickResult>");
            sb.append(getpickResult()[0]).append(",").append(getpickResult()[1]).append(",").append(getpickResult()[2]).append(",").append(getpickResult()[3]).append(",").append(getpickResult()[4]);
            sb.append("</pickResult>");

            sb.append(getUser().toXMLString());
            sb.append("</m>");
            //
            sb.append("</module>");

            return sb.toString();

    }

    /** 


     @return 
    */
    public final String getMatchResultXmlByRc()
    {

            StringBuilder sb = new StringBuilder();

            //
            sb.append("<room id='");
            sb.append("-1");
            sb.append("' name='");
            sb.append("noname");
            sb.append("' gamename='");
            sb.append(DdzLogic_Toac.name);
            sb.append("'>");

            //
            double tmpG;
            long tmpG2;
            String type = "";

            //
            long winG = Long.parseLong(getpickResult()[1]);

            //
            if (getpickResult()[0].equals("1"))
            {
                    type = "add";

                    //
                    tmpG = Math.floor(winG * DdzLogic_Toac.getcostG());
                    tmpG2 = Long.parseLong((new Double(tmpG)).toString());
                    winG = winG - tmpG2;

                    //
                    //每局花费存入
                    sb.append("<action type='add' id='");
                    sb.append(DdzLogic_Toac.getcostUid());
                    sb.append("' n='");
                    sb.append(DdzLogic_Toac.getcostUN());
                    sb.append("' g='");
                    sb.append((new Long(tmpG2)).toString());
                    sb.append("'/>");

            }
            else
            {
                    type = "sub";

                    //没有抽税
                    //输掉的金点由于没有赢家可以给，于是还给系统
                    sb.append("<action type='add' id='");
                    sb.append(DdzLogic_Toac.getcostUid());
                    sb.append("' n='");
                    sb.append(DdzLogic_Toac.getcostUN());
                    sb.append("' g='");

                    sb.append((new Long(winG)).toString());

                    sb.append("'/>");

            }


            sb.append("<action type='" + type + "' id='");
            sb.append(getUser().getId());
            sb.append("' n='");
            sb.append(getUser().getNickName());
            sb.append("' g='");

            sb.append((new Long(winG)).toString());

            sb.append("'/>");

            //
            sb.append("</room>");

            return sb.toString();

    }
    
}

File: CSharpServer/DdzServer/DdzServer/net/silverfoxserver/extmodel/RoomModelByToac.cs
/*
* SilverFoxServer: massive multiplayer game server for Flash, ...
* VERSION:3.0
* PUBLISH DATE:2015-9-2 
* GITHUB:github.com/wdmir/521266750_qq_com
* UPDATES AND DOCUMENTATION AT: http://www.silverfoxserver.net
* COPYRIGHT 2009-2015 SilverFoxServer.NET. All rights reserved.
* MAIL:521266750@qq.com
*/
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using net.silverfoxserver.core.util;
using net.silverfoxserver.core.model;

namespace DdzServer.net.silverfoxserver.extmodel
{
    public class RoomModelByToac
    {
        /// <summary>
        /// 
        /// </summary>
        private IUserModel _u;

        public IUserModel getUser()
        {
            return _u;            
        }

        /// <summary>
        /// 
        /// </summary>
        private string[] _pickResult = { "0", "0", "0" };

        /// <summary>
        /// winOrLost,fen,machine_pick
        /// </summary>
        public string[] pickResult {

            get {

                return _pickResult;
            
            }
        
        }

        /// <summary>
        /// 
        /// </summary>
        /// <param name="value"></param>
        public RoomModelByToac(IUserModel value)
        {
            _u = value.clone();        

            _pickResult = new string[]{ "0", "0", "0" };
        }

        /// <summary>
        /// 默认2倍，闪光为3倍
        /// 我压中的概率是三分之一
        /// </summary>
        /// <param name="selectInd_"></param>
        /// <param name="fen"></param>
        /// <returns></returns>
        public string[] pick_a_card(int selectInd_, Int64 jiaoFen)
        {
            //winOrLost,fen,machine_pick,your_pick,light
            //string[] pickResult = { "0", "0", "0" };
            _pickResult = new string[] { "0", "0", "0" , "0","0"};

            //
            Random r = new Random(RandomUtil.GetRandSeed());

            int machine_pick = r.Next(3);
            machine_pick++;

            int light = r.Next(2);

            //
            _pickResult[2] = machine_pick.ToString();
            _pickResult[3] = selectInd_.ToString();
            _pickResult[4] = light.ToString();

            //
            if (selectInd_ == machine_pick)
            {
                _pickResult[0] = "1";

                //获得的分值 = 押分;
                if (0 == light)
                {
                    _pickResult[1] = (jiaoFen * 1).ToString();
                }
                else
                {
                    _pickResult[1] = (jiaoFen * 2).ToString();                
                }

            }
            else
            {
                _pickResult[0] = "0";

                //扣掉的分值 = 押分 * 1
                _pickResult[1] = (jiaoFen * 1).ToString();
            }

            return _pickResult;
        }


        /// <summary>
        /// 
        /// </summary>
        /// <param name="n"></param>
        /// <param name="v"></param>
        /// <returns></returns>
        public string[] setVars(string n, string v)
        {

            string[] setStatus = { "false", "", "" };

            //
            string[] sp = v.Split(',');


            //selectInd
            int selectInd = Convert.ToInt32(sp[1]);
            Int64 jiaoFen = Convert.ToInt64(sp[2]);

            //
            if (1 != selectInd && 
                2 != selectInd && 
                3 != selectInd)
            {
                setStatus[0] = "false";
                setStatus[1] = "<status>1</status>";

                return setStatus;

            }

            //
            if (DdzLogic_Toac.g1 != jiaoFen &&
                DdzLogic_Toac.g2 != jiaoFen &&
                DdzLogic_Toac.g3 != jiaoFen)
            {
                setStatus[0] = "false";
                setStatus[1] = "<status>2</status>";

                return setStatus;
            }

            //---------------------------------------------------
            pick_a_card(selectInd, jiaoFen);

            //
            setStatus[0] = "true";
            setStatus[1] = _pickResult[0] + "," + _pickResult[1] + "," + _pickResult[2];
            
            return setStatus;
        }

        /// <summary>
        /// 
        /// </summary>
        /// <returns></returns>
        public string toXMLString()
        {

            StringBuilder sb = new StringBuilder();

            //
            sb.Append("<module>");
            sb.Append("<m n='");
            sb.Append(DdzLogic_Toac.name);
            sb.Append("'>");

            sb.Append("<pickResult>");
            sb.Append(pickResult[0] + "," + pickResult[1] + "," + pickResult[2] + "," + pickResult[3] + "," + pickResult[4]);
            sb.Append("</pickResult>");

            sb.Append(getUser().toXMLString());
            sb.Append("</m>");
            //
            sb.Append("</module>");

            return sb.ToString();

        }

        /// <summary>
        /// 
        /// </summary>
        /// <returns></returns>
        public string getMatchResultXmlByRc()
        {

            StringBuilder sb = new StringBuilder();

            //
            sb.Append("<room id='");
            sb.Append("-1");
            sb.Append("' name='");
            sb.Append("noname");
            sb.Append("' gamename='");
            sb.Append(DdzLogic_Toac.name);
            sb.Append("'>");

            //
            double tmpG;
            Int64 tmpG2;
            string type = string.Empty;

            //
            Int64 winG = Convert.ToInt64(pickResult[1]);

            //
            if ("1" == pickResult[0])
            {
                type = "add";

                //
                tmpG = Math.Floor(winG * DdzLogic_Toac.costG);
                tmpG2 = Int64.Parse(tmpG.ToString());
                winG = winG - tmpG2;

                //
                //每局花费存入
                sb.Append("<action type='add' id='");
                sb.Append(DdzLogic_Toac.costUid);
                sb.Append("' n='");
                sb.Append(DdzLogic_Toac.costUN);
                sb.Append("' g='");
                sb.Append(tmpG2.ToString());
                sb.Append("'/>");

            }
            else
            {
                type = "sub";

                //没有抽税
                //输掉的金点由于没有赢家可以给，于是还给系统
                sb.Append("<action type='add' id='");
                sb.Append(DdzLogic_Toac.costUid);
                sb.Append("' n='");
                sb.Append(DdzLogic_Toac.costUN);
                sb.Append("' g='");

                sb.Append(winG.ToString());

                sb.Append("'/>");

            }


            sb.Append("<action type='" + type + "' id='");
            sb.Append(getUser().Id);
            sb.Append("' n='");
            sb.Append(getUser().NickName);
            sb.Append("' g='");

            sb.Append(winG.ToString());

            sb.Append("'/>");

            //
            sb.Append("</room>");

            return sb.ToString();

        }

        
    }





}
